tick/hawkes/inference/tests/hawkes_cumulant_matching_test.py

```diff
@@ -8,212 +8,305 @@
+            penalty=None,
+            C=1e-3,
+            tol=1e-16,
+        )
+
+    def test_hawkes_cumulants_tf_solve_l1(self):
+        self._test_hawkes_cumulants_solve(
+            Learner=HawkesCumulantMatchingTf,
+            max_iter=8000,
+            step=1e-5,
+            solver='adam',
+            penalty='l1',
+            C=1e-6,
+            tol=1e-16,
+        )
+
+    @unittest.skip("pytorch not implemented yet")
+    def test_hawkes_cumulants_pyt_solve_l1(self):
+        self._test_hawkes_cumulants_solve(
+            Learner=HawkesCumulantMatchingPyT,
+            max_iter=4000,
+            print_every=30,
+            step=1e-4,
+            solver='adam',
+            penalty='l1',
+            C=1e-3,
+            tol=1e-16,
+        )
+
+    def test_hawkes_cumulants_tf_solve_l2(self):
+        self._test_hawkes_cumulants_solve(
+            Learner=HawkesCumulantMatchingTf,
+            max_iter=8000,
+            step=1e-5,
+            solver='adam',
+            penalty='l2',
+            C=1e-5,
+            tol=1e-16,
+        )
+
+    @unittest.skip("PyTorch yet to be implemented")
+    def test_hawkes_cumulants_pyt_solve_l2(self):
+        self._test_hawkes_cumulants_solve(
+            Learner=HawkesCumulantMatchingPyT,
+            max_iter=4000,
+            print_every=30,
+            step=1e-4,
+            solver='adam',
+            penalty='l2',
+            C=1e-3,
+            tol=1e-16,
+        )
+
+    def _test_hawkes_cumulants_solve(
+            self,
+            Learner=HawkesCumulantMatchingTf,
+            max_iter=4000,
+            print_every=30,
+            step=1e-4,
+            solver='adam',
+            penalty=None,
+            C=1e-3,
+            tol=1e-16,
+            verbose=False,
+
+    ):
+        """...Test that hawkes cumulant reached expected value
+        """
+        multi = Test.get_simulated_model()
+        n_nodes = multi.hawkes_simu.n_nodes
+        timestamps = multi.timestamps
+        baseline = multi.hawkes_simu.baseline
+        decays = multi.hawkes_simu.decays
+        integration_support = .3
+
+        learner = Learner(
+            integration_support=integration_support,
+            max_iter=max_iter,
+            print_every=print_every,
+            step=step,
+            solver=solver,
+            penalty=penalty,
+            C=C,
+            tol=tol,
+        )
+        learner.fit(timestamps)
+        if verbose:
+            learner.print_history()
+
+        expected_R_pred = np.linalg.inv(
+            np.eye(n_nodes) - multi.hawkes_simu.adjacency
+        )
+
+        if verbose:
+            print('\n')
+            print(f'expected_R_pred:\n{expected_R_pred}')
+            print(f'solution:\n{learner.solution}')
+
+        self.assertTrue(
+            np.allclose(
+                learner.solution,
+                expected_R_pred,
+                atol=0.1,  # TODO: explain why estimation is not so accurate
```

Comment:
@Mbompr Unfortunately, accuracy is not so good here. Are you familiar with the tuning of hyperparameters of `HawkesCumulantMatching`? If yes, can you please see if a better choice of `step`, `max_iter` etc... gives more accurate estimations? I can see from the paper that the accuracy was really good.

Reply:
Hi @Mbompr 

Could you look into this please?